components/client/src/main/java/com/hotels/styx/client/connectionpool/SimpleConnectionPool.java

```diff
@@ -191,6 +206,18 @@ public void connectionClosed(Connection connection) {
         availableConnections.remove(connection);
     }
 
+    @Override
+    public void close() {
+        active = false;
+        Connection con;
+        while ((con = availableConnections.poll()) != null) {
+            borrowedCount.decrementAndGet();
```

Comment:
Decrementing `borrowedCount` seems wrong. The connections in `availableConnections` are sitting idle in the pool, and not being borrowed.

Reply:
oh you are right @mikkokar ,  I added that by mistake...  Fixed.